src/main/scala/com.snowplowanalytics/iglu.schemaddl/migrations/Migration.scala

```diff
@@ -115,9 +118,96 @@ object Migration {
    * @return migration map of each Schema to list of all available migrations
    */
   def buildMigrationMap(schemas: List[IgluSchema]): MigrationMap =
-    buildMigrationMatrix(NonEmptyList.fromListUnsafe(schemas))
-      .map(source => (source.schemas.head.self, buildMigration(source)))
-      .groupBy(_._1)
-      .mapValues(_.map(_._2))
+    buildMigrationMatrix(NonEmptyList.fromListUnsafe(schemas)).right match {
```

Comment:
Can we switch `schemas` type to non-empty-list. I feel very unsecure about this. Also, I think `Ior` is a very good data type for the whole idea, but here we're just throwing away this information. Should we make it also `Ior` or throw it away in `buildMigrationMatrix`.

Reply:
I did not wanted to change initial function signature of `buildMigrationMap` but if you think it is okay to change, I think we can give non-empty list of schema to it and return `Ior`. They would be both good changes in my opinion.

Reply:
I'm really frustrated by amount of partial functions, like `fromListUnsafe` I left in the codebase. Schema DDL should contain only pure and total algorithms.